src/cargo/ops/cargo_rustc/mod.rs

```diff
@@ -599,8 +599,10 @@ fn build_base_args(cx: &Context,
         cmd.arg("-g");
     }
 
-    if ndebug {
-        cmd.args(&["--cfg", "ndebug"]);
+    if debug_assertions && opt_level > 0 {
```

Comment:
I wanted to make the output of `-v` at least somewhat prettier, although that may be a bit of a pipe dream at this point!